zcash_client_backend/src/data_api/error.rs

```diff
@@ -104,6 +108,7 @@ impl<N: fmt::Display> fmt::Display for Error<N> {
             Error::Builder(e) => write!(f, "{:?}", e),
             Error::Protobuf(e) => write!(f, "{}", e),
             Error::SaplingNotActive => write!(f, "Could not determine Sapling upgrade activation height."),
+            Error::MemoForbidden => write!(f, "It is not possible to send a memo to a transparent address."),
```

Comment:
It would be nice to have a test of trying to send a memo to a t-addr.

Reply:
Testing these spending functions is quite difficult right now because we don't have a good way of mocking up blockchain state that the wallet can spend from. When I had started this branch, I had also hoped to complete the in-memory wallet backend that would let me write such tests, but there has just been no opportunity to do so. The code for wallet transaction construction that this derives from has in fact *never* had any tests. I think we absolutely should add some, but that's not a level of effort I can put forth right now.

Reply:
I'll see if I can add some tests that handle this under the sqlite crate, but I would really like to not work through yet another weekend.

Reply:
Let's not work through the weekend!

Reply:
I've confirmed that the use case for this error is actually unreachable by construction, even though the types aren't able to reflect that fact: it is only produced by the `spend` method, and `TransactionRequest` construction fails if you try to construct a transaction request that violates this invariant.